stable/database/files/nuosm

```diff
@@ -268,6 +268,16 @@ function perform_restore() {
 
     # restore data and/or fix the metadata
     log "restoring archive and/or clearing restored archive physical metadata"
+
```

Comment:
What we are doing is replacing streaming which is only possible with download.  a download of archive needs twice the disk space.   A stream of backupset fails as nuodocker thinks it's an archive.   What we are doing here is using download.   If what is downloaded is an archive we rename it to be DB_DIR, and is in same state as if it was streamed.   If it's a backupset we continue which does a nuoarchive which uses hardlinks between the archive and backupset so it does not take more disk space.    it improves a download of archive, as nuodocker is a slow recursive copy for archive. it removes need for stream while maintaining the performance.

Reply:
> issue with nuodocker when wrong type

The `perform_restore()` function should work regardless of the backup type. For example, we supply a dummy backup type when performing database restore operation using the restore chart (see line 538).
I think supplying the backup type is optimization in the context of the `perform_restore()` function.

I'm a bit confused about what the issue is with `nuodocker`. The `nuodocker restore archive` is doing exactly the same as you added here (except it uses recursive copy instead of move). Can you paste the error that we get and the situation when it is seen?
I agree that `nuodocker` invocation is requiring more disk space so I see this change as a useful enhancement.

```python
        # Check if origin is just an archive
        if self._has_archive_data(origin_dir):
            origin_metadata_file = os.path.join(origin_dir, DockerCommands.ARCHIVE_METADATA_FILE)
            # Restored archive shouldn't have a metadata file. If one
            # exists in the origin dir and clean-metadata hasn't been
            # specified, error
            if os.path.exists(origin_metadata_file) and not clean_metadata:
                raise RuntimeError('Restoring archive {} failed. Existing metadata file {} found'.format(origin_dir, origin_metadata_file))

            if os.path.abspath(origin_dir) != os.path.abspath(restore_dir):
                if os.listdir(restore_dir):
                    raise RuntimeError('Restoring archive {} failed. Restore directory {} must be empty'.format(origin_dir, restore_dir))
                copy_tree(origin_dir, restore_dir)

            # Now remove the metadata file if it exists in the restore dir
            restore_metadata_file = os.path.join(restore_dir, DockerCommands.ARCHIVE_METADATA_FILE)
            if os.path.exists(restore_metadata_file):
                os.remove(restore_metadata_file)
...
```